### PR TITLE
[pt1][quant] Add fbgemm_qlinear_dynamic op

### PR DESCRIPTION
**Pull request by jianyuh**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #23330 [pt1][quant] Fix the dimension mismatch issues when running the BERT model
* #23212 [pt1][quant] Add the type matching rule for qconfig_dict
* #23128 [pt1][quant] Dynamic Quantized Linear operator and module
* **#23104 [pt1][quant] Add fbgemm_qlinear_dynamic op**

As suggested in https://github.com/pytorch/pytorch/pull/22891, we will add an overload for ```torch.fbgemm_linear_int8_weight``` (dynamic quantized version of linear function) that takes PackedLinearWeight as input and is pretty much the same in signature as regular aten::linear.

Differential Revision: [D16381552](https://our.internmc.facebook.com/intern/diff/D16381552/)

### CODE REVIEW

File: test/test_quantized.py [hunk 1]
```diff
@@ -273,6 +273,49 @@ def test_cat(self, X, num, axis, relu):
...
+        # Weight prepacking operator for dynamic quantized Linear
+        W_prepack = qlinear_prepack(W_q)
+        # Dynamic quantized Linear operator with prepacked weight
+        Y_fp32 = qlinear_dynamic(X_fp32, W_prepack, b_fp32)
+
+        Y_fp32_ref = F.linear(X_fp32, W_fp32, b_fp32)
+        if use_relu:
+            Y_fp32_ref[Y_fp32_ref < 0.0] = 0.0
+
+        torch.testing.assert_allclose(Y_fp32, Y_fp32_ref, rtol=0.0001, atol=1e-3)
```
Comment (raghuramank100) [inline test/test_quantized.py]: Can we have  a test that is exact: If you quantize and dequantize the activations prior to F.linear, can you get an exact match?

**Review by raghuramank100 — CHANGES_REQUESTED** [review]
Please address comments, thanks

File: test/test_quantized.py (hunk 1 above)
Comment (jianyuh) [reply test/test_quantized.py]: Right: currently we use a special example (the same as https://github.com/pytorch/pytorch/blob/master/test/test_jit.py#L11177-L11198), so we can have the exact match.

**Comment by jianyuh** [comment]
> Please address comments, thanks

Updated @raghuramank100 .

**Review by raghuramank100 — APPROVED** [review]
Please create a follow up task to add randomized testing for this op.

**Comment by jianyuh** [comment]
Created the task T47798162 for the unit test for the exact match between static and dynamic quantization.

**Comment by jianyuh** [comment]
> Please create a follow up task to add randomized testing for this op.

create a unit test for the exact match between static and dynamic quantization.

**Pull request closed without merging**